test/behavior/steps/login.py

```diff
@@ -0,0 +1,59 @@
+from behave import given, when, then
+
+
+@given('User is at the Tendrl login page')
+def step_impl(context):
+	context.browser.get(context.data['common']['tendrl']['ip'])
+
+
+@given('Tendrl API is not reachable')
+def step_impl(context):
+	pass
+
+
+@when('User types in a correct "{field}"')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).send_keys(context.data['common']['tendrl'][field])
+
+
+@when('User types in an incorrect "{field}"')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).send_keys("incorrect")
+
+
+@when('User types something in the "{field}" field')
+def step_impl(context, field):
+	context.browser.implicitly_wait(3)
+	context.browser.find_element_by_id(field).send_keys("something")
+
+
+@when('User deletes what is in the "{field}" field')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).clear()
+
+
+@when('User clicks the Login button')
+def step_impl(context):
+	context.browser.find_element_by_class_name("btn.btn-primary.btn-lg login-btn").click()
+
+
+@then('The text "username is required" should appear under the "username" box')
+def step_impl(context,):
+	context.browser.find_element_by_id("usernameHelpBlock")
+
+
+@then('The text "password is required" should appear under the "password" box')
+def step_impl(context):
+	context.browser.find_element_by_id("passwordHelpBlock")
+
+
+@then('The {message} popup should appear above the {field} box')
+def step_impl(context, message, field):
+	message = message[1:-1]
+	popup_text = context.browser.find_element_by_class_name("ng-binding").text
```

Comment:
create a unique class on id and use that. ng-binding is a generic class of angular.